src/amo/sagas/collections.js

```diff
@@ -0,0 +1,48 @@
+import { all, call, put, select, takeEvery } from 'redux-saga/effects';
+import {
+  FETCH_COLLECTION,
+  loadCollection,
+} from 'amo/reducers/collections';
+import * as api from 'amo/api/collections';
+import log from 'core/logger';
+import { createErrorHandler, getState } from 'core/sagas/utils';
+
+export function* fetchCollection({
+  payload: {
+    errorHandlerId,
+    page,
+    slug,
+    user,
+  },
+}) {
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  yield put(errorHandler.createClearingAction());
+
+  try {
+    const state = yield select(getState);
+
+    const { detail, addons } = yield all({
+      detail: call(api.getCollectionDetail, {
```

Comment:
It seems unnecessary to make two API calls. Could you file an issue requesting to add a `collection` parameter to the response of https://addons-server.readthedocs.io/en/latest/topics/api/collections.html#collection-add-ons-list ?

Reply:
Ah yes, I already talked to @eviljeff but did not open the issue yet. Will do!

Reply:
See: https://github.com/mozilla/addons-server/issues/6548.